homeassistant/components/duke_energy/__init__.py

```diff
@@ -0,0 +1,33 @@
+"""The Duke Energy integration."""
+
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .coordinator import DukeEnergyCoordinator
+
+# This integration provides no platforms for now, just inserts statistics
+PLATFORMS: list[Platform] = []
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Duke Energy from a config entry."""
+
+    coordinator = DukeEnergyCoordinator(hass, entry.data)
+    await coordinator.async_config_entry_first_refresh()
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
```

Comment:
Please extend config entry typing and store the coordinator in entry.runtime_data